#! /bin/sh
patch -p1 -l -f $* < $0
exit $?

From 41d046022e912c76cb424c906064745e732f01f3 Mon Sep 17 00:00:00 2001
From: Andre Heinecke <aheinecke@intevation.de>
Date: Wed, 30 Jul 2014 11:34:26 +0200
Subject: [PATCH] Set some accessibility information

    * qt4/main.cpp (qt_cmd_handler): Build buttons with accessibile
    Description.
    * qt4/pinentrydialog.cpp (setDescription, setError, setOkText)
    (setCancelText, setQualityBar): Set an accessible description.
    * qt4/pinentryconfirm.cpp (PinentryConfirm): Set message
    box contents also as accessible values.
    * NEWS: Mention it and the copy/paste change from last year.
---
 NEWS                    | 4 +++-
 qt4/main.cpp            | 5 ++++-
 qt4/pinentryconfirm.cpp | 2 ++
 qt4/pinentrydialog.cpp  | 8 +++++++-
 4 files changed, 16 insertions(+), 3 deletions(-)

diff --git a/NEWS b/NEWS
index a9c0057..fbc3708 100644
--- a/NEWS
+++ b/NEWS
@@ -1,7 +1,9 @@
 Noteworthy changes in version 0.8.4 (unreleased)
 ------------------------------------------------

- *
+ * Qt4: New option to enable pasting the passphrase from clipboard
+
+ * Qt4: Improved accessiblity


 Noteworthy changes in version 0.8.3 (2013-04-26)
diff --git a/qt4/main.cpp b/qt4/main.cpp
index 7fdef37..106999e 100644
--- a/qt4/main.cpp
+++ b/qt4/main.cpp
@@ -215,8 +215,11 @@ qt_cmd_handler (pinentry_t pe)
       };

       for ( size_t i = 0 ; i < sizeof buttonLabels / sizeof *buttonLabels ; ++i )
-        if ( (buttons & buttonLabels[i].button) && !buttonLabels[i].label.isEmpty() )
+        if ( (buttons & buttonLabels[i].button) && !buttonLabels[i].label.isEmpty() ) {
             box.button( buttonLabels[i].button )->setText( buttonLabels[i].label );
+            box.button( buttonLabels[i].button )->setAccessibleDescription ( buttonLabels[i].label );
+
+        }

       box.setIconPixmap( icon() );

diff --git a/qt4/pinentryconfirm.cpp b/qt4/pinentryconfirm.cpp
index b22aef5..f7a1d63 100644
--- a/qt4/pinentryconfirm.cpp
+++ b/qt4/pinentryconfirm.cpp
@@ -29,6 +29,8 @@ PinentryConfirm::PinentryConfirm(Icon icon, int timeout, const QString &title,
	connect(_timer, SIGNAL(timeout()), this, SLOT(slotTimeout()));
	_timer->start(timeout*1000);
     }
+    setAccessibleDescription (desc);
+    setAccessibleName (title);
 }

 void PinentryConfirm::slotTimeout()
diff --git a/qt4/pinentrydialog.cpp b/qt4/pinentrydialog.cpp
index a8dbb42..b16a74f 100644
--- a/qt4/pinentrydialog.cpp
+++ b/qt4/pinentrydialog.cpp
@@ -198,6 +198,7 @@ void PinEntryDialog::setDescription( const QString& txt )
 {
   _desc->setVisible( !txt.isEmpty() );
   _desc->setText( txt );
+  _desc->setAccessibleDescription ( txt );
   _icon->setPixmap( icon() );
   setError( QString::null );
 }
@@ -211,6 +212,7 @@ void PinEntryDialog::setError( const QString& txt )
 {
   if( !txt.isNull() )_icon->setPixmap( icon( QStyle::SP_MessageBoxCritical ) );
   _error->setText( txt );
+  _error->setAccessibleDescription ( txt );
   _error->setVisible( !txt.isEmpty() );
 }

@@ -243,19 +245,23 @@ QString PinEntryDialog::prompt() const
 void PinEntryDialog::setOkText( const QString& txt )
 {
   _ok->setText( txt );
+  _ok->setAccessibleDescription ( txt );
   _ok->setVisible( !txt.isEmpty() );
 }

 void PinEntryDialog::setCancelText( const QString& txt )
 {
   _cancel->setText( txt );
+  _cancel->setAccessibleDescription ( txt );
   _cancel->setVisible( !txt.isEmpty() );
 }

 void PinEntryDialog::setQualityBar( const QString& txt )
 {
-  if (_have_quality_bar)
+  if (_have_quality_bar) {
     _quality_bar_label->setText( txt );
+    _quality_bar_label->setAccessibleDescription ( txt );
+  }
 }

 void PinEntryDialog::setQualityBarTT( const QString& txt )
--
1.9.1
